### ISSUE TITLE
Base and notebook images have different sources for some libraries

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask-docker#36]
For example, in the base image python comes from `main`

```
$ docker run -it --rm daskdev/dask conda list python
...
+ conda list python
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
msgpack-python            0.5.6            py36h6bb024c_1
python                    3.6.5                hc3d631a_2
python-blosc              1.5.1            py36h14c3975_2
python-dateutil           2.7.3                    py36_0
```

But in the notebook image, it come from `conda-forge`.

```
docker run -it --rm daskdev/dask-notebook conda list python
...
+ conda list python
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
ipython                   6.5.0                    py36_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
msgpack-python            0.5.6            py36h2d50403_3    conda-forge
python                    3.6.5                         1    conda-forge
python-blosc              1.4.4                    py36_0    conda-forge
python-dateutil           2.7.3                      py_0    conda-forge
python-editor             1.0.3                    py36_0    conda-forge
python-oauth2             1.0.1                    py36_0    conda-forge
```

**Comment by danielfrg** [comment]
I think this is because the base image does `FROM continuumio/miniconda3:4.5.4
` and the notebook one comes `FROM jupyter/base-notebook`.

I would probably add conda-forge dependencies to the base img, what do people think?

**Comment by mrocklin** [comment]
I propose switching to defaults (for the libraries that we change) in https://github.com/dask/dask-docker/pull/37

This is a bit odd though, because there is still plenty of conda-forge libraries from the original install in Jupyter's `base-notebook`

**Comment by danielfrg** [comment]
Right, thats what i thought going for full conda-forge would be "easier" to get all the images more in sync.

Since we install numpy and pandas explicitly from the same place now I imagine it wont be an issue, those are the libs that I usually have problems between the two channels.

**Comment by ocefpaf** [comment]
> I'm taking a look at it now. Adding conda forge to base dependencies increases the size of the image from 650MB to 950MB daskdev/dask-worker 0.19.4 befb63f97d84 8 minutes ago 967MB daskdev/dask-worker 0.19.4-defaults 789fd3afdad1 12 minutes ago 654MB I'm looking at retroactively installing things from defaults onto the jupyter notebook image.. cc @ocefpaf

Thanks for the ping. I will need to take a closer look at the env to see the packages that may be causing this and how we can "cut the fat."

**Comment by mrocklin** [comment]
OK, I've switched to conda-forge for now, but it's pretty painful on image size

```
daskdev/dask             0.19.4              b00c41c7d47d        About a minute ago   1.15GB
daskdev/dask             0.19.4-defaults     51e1a809e886        3 minutes ago        590MB
```

I've raise a couple issue upstream on conda-forge feedstocks for pandas and openblas, but I suspect that the solution is more systemic here.  @ocefpaf if you're willing to look into this I would appreciate it.  Is there someone that you think I should ping on the Anaconda Inc side that would be easy to engage here (I suspect that you know more about those folks than I do).

I may also look into replicating the `jupyter/base-notebook` image with just the defaults channel as an alternative solution in parallel.  Currently it depends on some packages like `tini` and `jupyterhub` that are not in defaults.

**Comment by ocefpaf** [comment]
> I've raise a couple issue upstream on conda-forge feedstocks for pandas and openblas, but I suspect that the solution is more systemic here. @ocefpaf if you're willing to look into this I would appreciate it. Is there someone that you think I should ping on the Anaconda Inc side that would be easy to engage here (I suspect that you know more about those folks than I do).

I cannot promise to solve this soon but I'm interested in pursing lighter packages. Thanks for opening the issues and I'll investigate them ASAP. Some may be low hanging fruits, others may be buried deep in the dependencies.

**Comment by mrocklin** [comment]
No rush.  This is a longstanding issue.  Thanks for any time you're able to
spend here, but I have no expectations.

**Comment by jacobtomlinson** [comment]
This issue is reasonably old now and I think the problems here have been superseded by #161 anyway.

I'm going to close this out but happy to reopen if there is a desire to do so.
